tree-sitter-tips.md

```diff
@@ -0,0 +1,79 @@
+# Tree-sitter tips and tricks
+
+Language development using [Tree-Sitter](https://github.com/tree-sitter/tree-sitter) can be an arduous task. These tips are intended to help bring empiricism to the process by reducing trial and error (or at least introducing some guard rails).
```

Comment:
What do you think about phrasing this as "Grammar development" instead of "Language development"?

Another thought when I first read this is: should we host this kind of documentation on github/tree-sitter or github/tree-sitter-cli? I think tree-sitter-cli might be a better location as the API is a little higher level (you get to write grammars in JavaScript instead of JSON).